synapse/storage/databases/main/user_directory.py

```diff
@@ -238,15 +238,16 @@ def _get_next_batch(
 
                 # Update each user in the user directory.
                 for user_id, profile in users_with_profile.items():
-                    await self.update_profile_in_user_dir(
-                        user_id, profile.display_name, profile.avatar_url
-                    )
+                    if not await self.is_excluded_from_user_dir(user_id):
```

Comment:
it's probably fine, but for the record: what's the logic behind adding this here, and at line 353, where there was no check before?

Reply:
Consistency with `_handle_deltas` in the handlers code. 

Rebuilding the user directory from scratch (here) should have the same outcome as starting with an empty directory and building it by listening for membership events (`_handle_deltas`). Before my changes, the latter excluded support and deactivated users, so I think the former should too.